src/sentry/api/serializers/models/integration.py

```diff
@@ -119,16 +119,26 @@ def serialize(self, obj, attrs, user, organization_id=None):
         installation = obj.get_installation(organization_id)
 
         if self.action == 'link':
-            data['linkIssueConfig'] = installation.get_link_issue_config(
+            config = installation.get_link_issue_config(
                 self.group,
                 params=self.params,
             )
+            config = installation.merge_issue_config_defaults(
```

Comment:
since the jira form is dynamic and changes based on certain fields, i think this will break that behavior by using the default rather than getting the updated project from the params

Reply:
@macqueen check #9857 I have Jira overwrite the `merge_issue_config_defaults` so that it simply returns the config.